### PR TITLE
enhance(mayactl): add SC and access mode information in volume list

### PR DESCRIPTION
**Pull request by ashishranjan738**
This PR does the following things.
-> Enhances CAS Engine to list PersistentVolume which will helpful in listing information of persistent volumes using Runtask.
-> Adds Runtask to fetch storageclass and access mode information for cstor and jiva which will helpful for clients that depends on m-apiserver for information related to volumes.
-> Adds access mode and storageclass information in volume list API which can be consumed by clients like mayactl.
-> Adds StorageClass and access mode information in mayactl ```volume list``` command which will be useful for users.
-> Output of `mayactl volume list`:
```
Namespace  Name                                      Status   Type   Capacity  Storage Class         Access Mode
---------  ----                                      ------   ----   --------  -------------         -----------
default    pvc-c530cae3-140c-11e9-9348-b4b686bd0cff  Running  jiva   5G        openebs-jiva-default  ReadWriteOnce
openebs    pvc-c53033e9-140c-11e9-9348-b4b686bd0cff  Running  cstor  5G        openebs-cstor-sparse  ReadWriteOnce
``` 
-> Output of response from list endpoint:
```
{
  "items": [
    {
      "apiVersion": "v1alpha1",
      "kind": "CASVolume",
      "metadata": {
        "annotations": {
          "vsm.openebs.io/replica-status": "running",
          "openebs.io/controller-ips": "172.17.0.8",
          "openebs.io/iqn": "iqn.2016-09.com.openebs.jiva:pvc-09158135-149c-11e9-a935-b4b686bd0cff",
          "openebs.io/replica-count": "1",
          "openebs.io/replica-ips": "172.17.0.7",
          "vsm.openebs.io/cluster-ips": "10.0.0.153",
          "openebs.io/storage-class": "openebs-jiva-default",
          "openebs.io/targetportals": "10.0.0.153:3260",
          "vsm.openebs.io/iqn": "iqn.2016-09.com.openebs.jiva:pvc-09158135-149c-11e9-a935-b4b686bd0cff",
          "vsm.openebs.io/replica-count": "1",
          "openebs.io/controller-status": "running,running",
          "openebs.io/volume-size": "5G",
          "vsm.openebs.io/controller-ips": "172.17.0.8",
          "vsm.openebs.io/volume-size": "5G",
          "openebs.io/cluster-ips": "10.0.0.153",
          "openebs.io/replica-status": "running",
          "vsm.openebs.io/controller-status": "running,running",
          "vsm.openebs.io/replica-ips": "172.17.0.7",
          "vsm.openebs.io/targetportals": "10.0.0.153:3260"
        },
        "name": "pvc-09158135-149c-11e9-a935-b4b686bd0cff",
        "namespace": "default"
      },
      "spec": {
        "accessMode": "ReadWriteOnce",
        "capacity": "5G",
        "casType": "jiva",
        "fsType": "",
        "iqn": "iqn.2016-09.com.openebs.jiva:pvc-09158135-149c-11e9-a935-b4b686bd0cff",
        "lun": 0,
        "replicas": "1",
        "targetIP": "10.0.0.153",
        "targetPort": "3260",
        "targetPortal": "10.0.0.153:3260"
      },
      "status": {
        "Message": "",
        "Phase": "",
        "Reason": ""
      }
    },
    {
      "apiVersion": "v1alpha1",
      "kind": "CASVolume",
      "metadata": {
        "annotations": {
          "openebs.io/storage-class": "openebs-cstor-sparse",
          "openebs.io/volume-size": "5G",
          "openebs.io/cluster-ips": "10.0.0.135",
          "openebs.io/controller-status": "running running running"
        },
        "name": "pvc-09151006-149c-11e9-a935-b4b686bd0cff",
        "namespace": "openebs"
      },
      "spec": {
        "accessMode": "ReadWriteOnce",
        "capacity": "5G",
        "casType": "cstor",
        "fsType": "",
        "iqn": "iqn.2016-09.com.openebs.cstor:pvc-09151006-149c-11e9-a935-b4b686bd0cff",
        "lun": 0,
        "replicas": "1",
        "targetIP": "10.0.0.135",
        "targetPort": "3260",
        "targetPortal": "10.0.0.135:3260"
      },
      "status": {
        "Message": "",
        "Phase": "",
        "Reason": ""
      }
    }
  ],
  "metadata": {},
  "metalist": {}
}

```

The above enhancements are good to have in openebs.

**Which issue this PR fixes** : 
https://github.com/openebs/openebs/issues/1981
https://github.com/openebs/openebs/issues/1994

**Special notes for your reviewer**:
-> If the values are not present in the response the default value will be "<none>".

### CODE REVIEW

File: pkg/apis/openebs.io/v1alpha1/cas_volume.go [hunk 1]
```diff
@@ -119,6 +119,8 @@ type CASVolumeSpec struct {
 	FSType string `json:"fsType"`
 	// Lun will specify the lun number 0, 1.. on iSCSI Volume. (default: 0)
 	Lun int32 `json:"lun"`
+	// AccessMode will hold the access mode of this volume
```
Comment (prateekpandey14) [inline pkg/apis/openebs.io/v1alpha1/cas_volume.go]: nit: access mode of a volume

File: pkg/install/v1alpha1/cstor_volume.go [hunk 2]
```diff
@@ -685,13 +703,16 @@ spec:
...
+    {{- $accessMode :=  pluck "accessModes" $map | first }}
+    {{- $storageClass := pluck "storageClass" $map | first }}
     {{- $name := $pkey }}
       - kind: CASVolume
         apiVersion: v1alpha1
         metadata:
           name: {{ $name }}
           namespace: {{ $namespace }}
           annotations:
+            openebs.io/storage-class: {{ $storageClass }}
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply pkg/install/v1alpha1/cstor_volume.go]: addressed

File: pkg/apis/openebs.io/v1alpha1/cas_volume.go (hunk 1 above)
Comment (ashishranjan738) [reply pkg/apis/openebs.io/v1alpha1/cas_volume.go]: done

File: cmd/mayactl/app/command/volumes_list.go [hunk 3]
```diff
@@ -63,15 +63,15 @@ func (c *CmdVolumeOptions) RunVolumesList(cmd *cobra.Command) error {
 	}
 
 	out := make([]string, len(cvols.Items)+2)
-	out[0] = "Namespace|Name|Status|Type|Capacity"
-	out[1] = "---------|----|------|----|--------"
-	for i, items := range cvols.Items {
-		if len(items.Status.Reason) == 0 {
-			items.Status.Reason = volumeStatusOK
+	out[0] = "Namespace|Name|Status|Type|Capacity|Storage Class|Access Mode"
```
Comment (prateekpandey14) [inline cmd/mayactl/app/command/volumes_list.go]: can we `StorageClass` ?

**Pull request merged**